main.tf

```diff
@@ -0,0 +1,143 @@
+  region                  = join("", split("-", var.region))
+  class_env               = var.environment
+  cloud_resource_type     = each.value.name
+  instance_env            = var.environment_number
+  instance_resource       = var.resource_number
+  maximum_length          = each.value.max_length
+  use_azure_region_abbr   = true
+
+}
+
+module "resource_group" {
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-resource_group.git?ref=1.0.0"
+
+  location = var.region
+  name     = module.resource_names["resource_group"].standard
+
+  tags = merge(var.tags, { resource_name = module.resource_names["resource_group"].standard })
+}
+
+module "public_ip" {
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-public_ip?ref=1.0.0"
+
+  name                = module.resource_names["public_ip"].standard
+  resource_group_name = module.resource_group.name
+  location            = var.region
+  allocation_method   = "Static"
+  domain_name_label   = module.resource_names["public_ip"].standard
+  sku                 = "Standard"
+  sku_tier            = "Regional"
+  zones               = var.zones
+
+  tags = merge(var.tags, {
+    resource_name = module.resource_names["public_ip"].standard
+  })
+
+  depends_on = [module.resource_group]
+}
+
+module "managed_identity" {
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-user_managed_identity.git?ref=1.0.0"
+
+  count = var.create_user_managed_identity ? 1 : 0
+
+  resource_group_name         = module.resource_group.name
+  location                    = var.region
+  user_assigned_identity_name = module.resource_names["msi"].standard
+
+  depends_on = [module.resource_group]
+}
+
+module "identity_roles" {
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-role_assignment.git?ref=1.0.0"
+
+  for_each = var.create_user_managed_identity ? var.role_assignments : {}
+
+  principal_id         = module.managed_identity[0].principal_id
+  role_definition_name = each.value[0]
+  scope                = each.value[1]
+
+  depends_on = [module.managed_identity, module.resource_group]
+}
+
+module "private_dns_record" {
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-private_dns_records.git?ref=1.0.0"
+
+  count = var.appgw_private && var.custom_private_dns_record != null ? 1 : 0
+
+  a_records = {
+    app_gateway = {
+      name                = var.custom_private_dns_record["name"]
+      zone_name           = var.custom_private_dns_record["private_dns_zone_name"]
+      resource_group_name = var.custom_private_dns_record["private_dns_zone_rg"]
+      ttl                 = var.custom_private_dns_record["ttl"]
+      records             = [var.private_ip_address]
+    }
+  }
+}
+
+module "application_gateway" {
+  source = "git::https://github.com/launchbynttdata/tf-azurerm-module_primitive-application_gateway.git?ref=feature/init"
```

Comment:
Needs to be pointed at a tag, pretty sure this is causing your lint failure as this branch no longer exists.

Reply:
done. sorry about that